ddbToEsDlqLambda/index.ts

```diff
@@ -0,0 +1,161 @@
+            QueueUrl: queueUrl,
+            MaxNumberOfMessages: batchSize,
+            VisibilityTimeout: MESSAGE_VISIBILITY_TIMEOUT,
+            WaitTimeSeconds: 0,
+        })
+        .promise()
+        .catch((e) => {
+            throw new Error(`Failed to receive messages: ${JSON.stringify(e)}`);
+        });
+
+    // when there are no messages, resp.Messages is undefined
+    const messages = resp.Messages === undefined ? [] : resp.Messages;
+    logger.debug('Received messages', JSON.stringify(messages));
+    return messages;
+};
+
+const deleteMessages = async (queueUrl, messagesToDelete) => {
+    const resp = await sqs
+        .deleteMessageBatch({
+            QueueUrl: queueUrl,
+            Entries: messagesToDelete,
+        })
+        .promise()
+        .catch((e) => {
+            throw new Error(`Failed to delete message: ${JSON.stringify(e)}`);
+        });
+
+    // resp.Successful contains successfully deleted messages.
+    if (resp.Successful) {
+        logger.debug('Deleted messages', JSON.stringify(resp.Successful));
+    }
+
+    // resp.Failed contains messages that were failed to delete
+    if (resp.Failed && resp.Failed.length) {
+        throw new Error(`Failed to delete messages: ${JSON.stringify(resp.Failed)}`);
+    }
+};
+
+/**
+ * This is asynchronous invocation, the successful invocation doesn't mean
+ * successful function execution.
+ */
+const invokeDdbToEsLambda = async (records) => {
+    const resp = await lambda
+        .invoke({
+            FunctionName: DDB_TO_ES_LAMBDA_NAME,
+            InvocationType: 'Event',
+            Payload: JSON.stringify(records),
+        })
+        .promise()
+        .catch((e) => {
+            throw new Error(`Failed to invoke DdbToEsLambda: ${JSON.stringify(e)}`);
+        });
+
+    if (resp.StatusCode >= 400) {
+        throw new Error(`Failed to invoke DdbToEsLambda: ${JSON.stringify(resp.Payload)}`);
+    }
+
+    logger.debug('Invoked DdbToEsLambda with records', JSON.stringify(records));
+};
+
+const getRecordsFromDdbStream = async (message) => {
+    const resp = await dynamodbstreams
+        .getShardIterator({
+            ShardId: message.DDBStreamBatchInfo.shardId,
+            ShardIteratorType: 'AT_SEQUENCE_NUMBER',
+            StreamArn: message.DDBStreamBatchInfo.streamArn,
+            SequenceNumber: message.DDBStreamBatchInfo.startSequenceNumber,
+        })
+        .promise()
+        .catch((e) => {
+            throw new Error(`Failed to get shard iterator: ${JSON.stringify(e)}`);
+        });
+
+    const records = await dynamodbstreams
+        .getRecords({
+            ShardIterator: resp.ShardIterator,
+        })
+        .promise()
+        .catch((e) => {
+            throw new Error(`Failed to get records: ${JSON.stringify(e)}`);
+        });
+
+    logger.debug('Fetched records', JSON.stringify(records));
+    return records;
+};
+
+/**
+ * This lambda function would read messages form DbToEs DLQ, fetch records
```

Comment:
A few minor suggestions to your description:

```suggestion
 * This Lambda function will read message from the DDbToEs DLQ, fetch records from the DynamoDB stream referenced in the queued message, and invoke the DDBToEs Lambda function providing the records as input.
```